dvc/remote/base.py

```diff
@@ -36,31 +36,21 @@ def __init__(self, cmd, ret, err):
         super(RemoteBaseCmdError, self).__init__(m.format(cmd, ret, err))
 
 
+class RemoteMissingDepsError(DvcException):
+    pass
+
+
 class RemoteBase(object):
     REGEX = None
     REQUIRES = {}
     JOBS = 4 * cpu_count()
 
     def __init__(self, project, config):
-        pass
-
-    def __repr__(self):
-        return "{class_name}: '{url}'".format(
-            class_name=type(self).__name__,
-            url=(self.url or 'No url')
-        )
-
-    def compat_config(config):
-        return config.copy()
-
-    @classmethod
-    def supported(cls, config):
-        url = config[Config.SECTION_REMOTE_URL]
-        url_ok = cls.match(url) is not None
-        deps_ok = all(cls.REQUIRES.values())
-        if url_ok and not deps_ok:
-            missing = [k for k, v in cls.REQUIRES.items() if v is None]
-            logger.warning(
+        self.project = project
+        deps_ok = all(self.REQUIRES.values())
+        if not deps_ok:
```

Comment:
@efiop , do you think we could add something similar for "Dependencies" to address this issue: https://github.com/iterative/dvc/issues/1443 ?